### PR TITLE
Fix seeking in replays not correctly pausing samples

### PR DESCRIPTION
**Pull request by peppy**
This was a complete oversight. Wasn't being cached and wasn't being updated correctly with the new bindable flow. I've removed the `IsSeeking` as it was only ever used for this purpose, replacing with a common method that should make it easier to understand its purpose in the future.

Comes with test coverage (slightly more comprehensive compared to editor version).

### CODE REVIEW

**Comment by peppy** [comment]
This is broken due to caching precedence (editor is not getting the correct pause state due to local caching added in `FrameStabilityContainer`).

**Comment by peppy** [comment]
I've fixed this in the only simple way I can think of. Hopefully we can agree on going with this short term to fix the issue. Open to other suggestions that don't involve changing the hierarchy massively.

File: osu.Game/Rulesets/UI/FrameStabilityContainer.cs [hunk 1]
```diff
@@ -207,11 +211,15 @@ private void updateClock()
 
         private void setClock()
         {
-            // in case a parent gameplay clock isn't available, just use the parent clock.
-            parentGameplayClock ??= Clock;
-
-            Clock = GameplayClock;
-            ProcessCustomClock = false;
```
Comment (smoogipoo) [inline osu.Game/Rulesets/UI/FrameStabilityContainer.cs:214]: Sure you want to omit this?

Are these changes still required at all?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Rulesets/UI/FrameStabilityContainer.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Rulesets/UI/FrameStabilityContainer.cs:214]: yes, see the ProcessFrame method.

**Review by smoogipoo — APPROVED** [review]
